src/main/java/com/wolox/training/exception/ErrorHandler.java

```diff
@@ -0,0 +1,112 @@
+package com.wolox.training.exception;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+import javax.validation.ConstraintViolation;
+import javax.validation.ConstraintViolationException;
+import java.sql.Timestamp;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@ControllerAdvice
+public class ErrorHandler extends ResponseEntityExceptionHandler {
+
+    @ExceptionHandler({
+            BookNotFoundException.class
+    })
+    public ResponseEntity<Object> handleNotFound(Exception ex, WebRequest request) {
```

Comment:
Could you specify here what to return as the error response? In order to avoid Object.

Reply:
I ve change the error handler for return specific ResponseEntity instead "Object".